src/sentry/static/sentry/app/stores/projectsStore.jsx

```diff
@@ -37,23 +41,29 @@ const ProjectsStore = Reflux.createStore({
 
     this.itemsById = {
       ...this.itemsById,
+      [newProject.id]: newProject,
     };
 
-    this.itemsById[newProject.id] = newProject;
-
     // Ideally we'd always trigger this.itemsById, but following existing patterns
     // so we don't break things
     this.trigger(new Set([prevProject.id]));
   },
 
   onCreateSuccess(project) {
-    this.itemsById[project.id] = project;
+    this.itemsById = {
+      ...this.itemsById,
+      [project.id]: project,
+    };
     this.trigger(new Set([project.id]));
   },
 
   onUpdateSuccess(data) {
     let project = this.getById(data.id);
-    Object.assign(project, data);
+    let newProject = Object.assign({}, project, data);
+    this.itemsById = {
```

Comment:
Just realizing now that treating `itemsById` as immutable here is probably not needed because I don't think we actually ever access it directly. Where this *can* lead to problems is if we depend on behavior when using `this.trigger(this.itemsById)`.